test/nginx.conf

```diff
@@ -4,29 +4,74 @@ events {
-        add_header 'Access-Control-Allow-Headers' 'Authorization,Content-Type,Accept,Origin,User-Agent,DNT,Cache-Control,X-Mx-ReqToken,Keep-Alive,X-Requested-With,If-Modified-Since' always;
-        add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range' always;
-        add_header 'Access-Control-Max-Age' 1728000; # cache preflight value for 20 days
-
-        # Alias the regexps, to ensure that they're not rewritten.
-        set $room_id $2;
-        set $event_id $3;
-        proxy_pass http://127.0.0.1:8082/api/1/report/$room_id/$event_id;
+        listen [::]:8081 ipv6only=off;
+
+        location ~ ^/_matrix/client/(r0|v3)/rooms/([^/\s]*)/report/(.*)$ {
+            # Abuse reports should be sent to Mjölnir.
+            # The r0 endpoint is deprecated but still used by many clients.
+            # As of this writing, the v3 endpoint is the up-to-date version.
+
+            # Add CORS, otherwise a browser will refuse this request.
+            add_header 'Access-Control-Allow-Origin' '*' always; # Note: '*' is for testing purposes. For your own server, you probably want to tighten this.
+            add_header 'Access-Control-Allow-Credentials' 'true' always;
+            add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS' always;
+            add_header 'Access-Control-Allow-Headers' 'Authorization,Content-Type,Accept,Origin,User-Agent,DNT,Cache-Control,X-Mx-ReqToken,Keep-Alive,X-Requested-With,If-Modified-Since' always;
+            add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range' always;
+            add_header 'Access-Control-Max-Age' 1728000; # cache preflight value for 20 days
+
+            # Alias the regexps, to ensure that they're not rewritten.
+            set $room_id $2;
+            set $event_id $3;
+            proxy_pass http://127.0.0.1:8082/api/1/report/$room_id/$event_id;
+        }
+        # MSC1929 Test in best-case
+        location /.well-known/matrix/support {
+            default_type application/json;
+            return 200 '{
+                "contacts": [{
+                    "matrix_id": "@admin:localhost",
+                    "role": "admin"
+                }],
+                "support_page": "http://localhost"
+            }';
+        }
+        location / {
+            # Everything else should be sent to Synapse.
+            proxy_pass http://127.0.0.1:9999;
+        }
+    }
+    server {
+        listen [::]:7070 ipv6only=off;
+
+        # MSC1929 Test in worst-case
+        location /.well-known/matrix/support {
+            default_type application/json;
+            return 404 '{}';
+        }
+    }
+    server {
+        listen [::]:7071 ipv6only=off;
+
+        # MSC1929 Test in supportpage_only-case
+        location /.well-known/matrix/support {
+            default_type application/json;
+            return 200 '{
+                "support_page": "http://localhost"
```

Comment:
Can you explain what you changed in this file? From what I can gather it's just the same as before but with some test well knowns?

Reply:
It adds the test cases for the well-known stuff and fixes the indentation of the old stuff. The indentation was off by one essentially.

It adds multiple files along different ports as the path obviously is hardcoded. So to have multiple test cases I split it across multiple ports instead.